### PR TITLE
www-misc/profile-sync-daemon: add masked 6.28 and 9999 version

### PR DESCRIPTION
**Pull request by soredake**
Added new masked systemd-only version.
Eapi bump on older version.

### CODE REVIEW

File: profiles/package.mask [hunk 1]
```diff
@@ -30,6 +30,11 @@
 
 #--- END OF EXAMPLES ---
 
+# Since 6.0, Profile Sync Daemon developers removed all non-systemd 
+support, so it won't be compatible with any other init that's not systemd.
```
Comment (mgorny) [inline profiles/package.mask]: You've got missing `#` here.

**Comment by mgorny** [comment]
FYI, the malformed p.mask killed pkgcheck, so please fix it and push over to get QA check results.

**Comment by soredake** [comment]
Oops, fixed.

**Comment by doughdemon** [comment]
Is the eapi bump of the older (stable!) version intentionally included in this PR?

**Comment by soredake** [comment]
Yes.

**Comment by gktrk** [comment]
@gentoo/proxy-maint

**Comment by soredake** [comment]
ping

**Comment by gktrk** [comment]
I don't think masking is the right approach here. It's better to check REPLACING_VERSIONS and ewarn about openrc support being dropped. We can't keep the versions <6 in the tree forever, so masking only delays the inevitable.

**Comment by gktrk** [comment]
Also, seems like [/etc/psd.conf is no longer supported](https://github.com/graysky2/profile-sync-daemon/blob/master/common/profile-sync-daemon.in#L187). We should have an einfo to that effect in pkg_postinst along with openrc support being dropped.

**Comment by soredake** [comment]
Commit updated.

File: profiles/package.mask [hunk 2]
```diff
@@ -208,7 +208,7 @@ media-gfx/skencil
 net-wireless/chillispot
 
 # Pacho Ramos <pacho@gentoo.org> (08 Oct 2016)
-# Fails to build (#515294), nothing needs it, relies on obsolete capi4kutils. 
+# Fails to build (#515294), nothing needs it, relies on obsolete capi4kutils.
```
Comment (gktrk) [inline profiles/package.mask]: Erroneous whitespace here.

File: www-misc/profile-sync-daemon/profile-sync-daemon-5.45.1.ebuild [hunk 3]
```diff
@@ -1,12 +1,12 @@
-# Copyright 1999-2014 Gentoo Foundation
+# Copyright 1999-2016 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=5
+EAPI=6
 
-inherit eutils vcs-snapshot
+inherit eutils
```
Comment (gktrk) [inline www-misc/profile-sync-daemon/profile-sync-daemon-5.45.1.ebuild]: I don't see any reason to inherit eutils. Am I missing something?

File: www-misc/profile-sync-daemon/profile-sync-daemon-6.28.ebuild [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+RDEPEND="
+	app-shells/bash
+	net-misc/rsync[xattr]
+	sys-apps/systemd"
+
+pkg_postinst() {
+	local replacing
+	for replacing in ${REPLACING_VERSIONS}; do
+		if [[ "$(get_major_version ${replacing})" == "5" ]]; then
```
Comment (gktrk) [inline www-misc/profile-sync-daemon/profile-sync-daemon-6.28.ebuild]: `get_major_version` is provided by the versionator eclass, so it should be inherited properly.

**Comment by soredake** [comment]
Rebased and updated.

File: www-misc/profile-sync-daemon/profile-sync-daemon-6.28.ebuild [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+EAPI=6
+
+inherit versionator
+
+DESCRIPTION="Symlinks and syncs browser profile dirs to RAM."
+HOMEPAGE="https://wiki.archlinux.org/index.php/Profile-sync-daemon"
+
+SRC_URI="https://github.com/graysky2/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+KEYWORDS="~amd64 ~x86"
+RESTRICT="mirror"
```
Comment (soredake) [inline www-misc/profile-sync-daemon/profile-sync-daemon-6.28.ebuild]: Restrict removed.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/bab1492...9e31234. Thanks!

**Pull request closed without merging**